### PR TITLE
RavenDB-14585 Allow to define migrator path in settings.json

### PR DESCRIPTION
**Pull request by iftahbe**

### CODE REVIEW

File: src/Raven.Server/Config/Categories/MigratorConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+using Raven.Server.Config.Settings;
+
+namespace Raven.Server.Config.Categories
+{
+    public class MigratorConfiguration : ConfigurationCategory
+    {
+        [Description("The full path of the directory containing the Raven.Migrator executable. Setting this option here will disable the ability to set a path in the Studio.")]
+        [DefaultValue(null)]
+        [ConfigurationEntry("Migrator.FullPath", ConfigurationEntryScope.ServerWideOnly)]
+        public PathSetting FullPath { get; set; }
```
Comment (ppekrol) [inline src/Raven.Server/Config/Categories/MigratorConfiguration.cs]: [ConfigurationEntry("Migration.MigratorPath", ConfigurationEntryScope.ServerWideOnly)]

File: src/Raven.Server/Config/Categories/MigratorConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel;
+using Raven.Server.Config.Attributes;
+using Raven.Server.Config.Settings;
+
+namespace Raven.Server.Config.Categories
+{
+    public class MigratorConfiguration : ConfigurationCategory
```
Comment (ppekrol) [inline src/Raven.Server/Config/Categories/MigratorConfiguration.cs]: MigrationConfiguration

File: src/Raven.Server/Config/Categories/MigratorConfiguration.cs (hunk 1 above)
Comment (ppekrol) [inline src/Raven.Server/Config/Categories/MigratorConfiguration.cs]: Property name => MigratorPath

File: src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs [hunk 3]
```diff
@@ -464,15 +465,36 @@ public Task GetMigratedServerUrls()
...
+        [RavenAction("/admin/smuggler/migrator-path", "GET", AuthorizationStatus.Operator)]
+        public Task GetConfigurationMigratorPath()
+        {
+            // If the path from the configuration is defined, the Studio will block the option to set the path in the import view
+            using (Database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    writer.WriteStartObject();
+                    writer.WritePropertyName("PathDefined");
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs]: nameof(MigrationConfiguration.MigratorPath)

File: src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs [hunk 4]
```diff
@@ -464,15 +465,36 @@ public Task GetMigratedServerUrls()
             return Task.CompletedTask;
         }
 
+        [RavenAction("/admin/smuggler/migrator-path", "GET", AuthorizationStatus.Operator)]
+        public Task GetConfigurationMigratorPath()
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs]: Isn't this more like a StudioTaskHandler endpoint? This is used purely for Studio, right?

Is Operator fine (I think it is). Who can do migration?

File: src/Raven.Server/Config/RavenConfiguration.cs [hunk 5]
```diff
@@ -80,6 +80,8 @@ public class RavenConfiguration
 
         public UpdatesConfiguration Updates { get; }
 
+        public MigratorConfiguration Migrator { get; }
```
Comment (ppekrol) [inline src/Raven.Server/Config/RavenConfiguration.cs]: Migration

File: src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs (hunk 4 above)
Comment (iftahbe) [reply src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs]: Yes, Studio only. I'll move it.
Migration can be done by Operator.

File: src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs (hunk 3 above)
Comment (iftahbe) [reply src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs]: I think we don't want to expose the actual path, only if it is defined or not.
So I'm sending a boolean here.

**Pull request merged**